### DISCUSSION TITLE
cdk synth extremely slow

### DISCUSSION THREAD

**Discussion by benm5678** [aws/aws-cdk#25837] [General]
What could possible cause cdk synth to be extremely slow in particular project?
It's similar size as other projects we have, yet synth can take up to 15min (no CPU overload & can't see network traffic) -- the files in cdk.out get added very slowly.
Other similar projects on that same PC work fine.
The "slow" project runs synth fast on another developer's PC (~20sec).  But on PC it's slow on, even doing git clone to a fresh dir still doesn't help.  Adding "--verbose" does not provide any helpful info... we see it stuck on "STARTING deployment for: "

**Comment by manmartgarc** [comment]
are you on WSL by any chance

**Reply by soderluk** [reply]
@manmartgarc I am, and suddenly the `cdk synth` takes a long time. Running my tests with pytest takes over 40s, and I found out that the culprit is most likely the CDK stack tests. Was there something in regards of WSL that you had in mind?

**Reply by manmartgarc** [reply]
Yes, so [WSL2 is known](https://learn.microsoft.com/en-us/windows/wsl/compare-versions) to have worse IO performance when accessing cross-OS files (read C:/ from WSL or vice versa) than WSL1. The solution is to use the Linux file system directly. If you're already doing this, then it's probably something else.

Obviously the synth duration is a function of what you're synthesizing; if you have a gigantic container that needs to be built and shipped, well there's no avoiding that.

---

One of the common issues w/ CDK tests is that you might be synthesizing the same stack multiple times.

**Reply by soderluk** [reply]
@manmartgarc Yeah, I'm only running on the WSL without accessing the Windows filesystem from WSL. I found out the culprit actually being the code coverage, when using it with Python 3.12. Setting the environment variable `COVERAGE_CORE=sysmon` when using Python 3.12 brought down the tests time by 50%.
Thank you for your answer anyway!

**Reply by manmartgarc** [reply]
Interesting! Glad it worked out. Another thing is to make sure your coverage is not scanning over the generated CFN templates or anything other than what you intend to cover
